test/index.ts

```diff
@@ -447,3 +449,10 @@ function testXOR() {
   type Test1 = Assert<IsExact<Actual1, Expected1>>;
   type Test2 = Assert<IsExact<Actual2, Expected2>>;
 }
+
+function testNoop() {
+  const x: void = noop();
+  const y: void = noop(false, 0, "", {}, [], null, undefined, Promise.resolve(), new Error(), noop);
+  const z: Promise<void> = Promise.resolve("foo").then(erase);
+  const callback: (_: number) => void = Math.random() < 0.5 ? (i => void i) : noop;
```

Comment:
the `void` operator is weird ;)
```suggestion
  const callback: (_: number) => void = Math.random() < 0.5 ? (() => {}) : noop;
```

Reply:
I always find `{}` dubious since it looks like an empty code block or an object literal at the same time, but sure

Reply:
Also, I guess if this file isn't meant to serve as documentation, the `i` callback for assignment equality showcasing isn't useful